### DISCUSSION TITLE
Sidekiq in Adonis js

### DISCUSSION THREAD

**Discussion by KhanAmaan55** [adonisjs/core#4215] [Help] [answered]
Can we use Sidekiq in Adonis js?
is there any built-in support for it?

**Comment by McSneaky** [comment]
No built in support

I can't find any community package for it currently either

You can still use it manually tho. Install any sidekiq client you want and use it like you'd use in any other Node project

I haven't used it myself and couldn't find any up-to-date Node clients. Seems like most of them are rather old

**Comment by gauravds** [comment]
Sidekiq is the rails version, I don't think there is available any such library to support it and poll jobs from there. But some kind of version of a similar application is https://github.com/Rocketseat/adonis-bull/tree/alpha, but this is also not maintained for the last two years. But it seems fine on my production code.

**Comment by MaximeMRF** [comment, marked as answer]
Hello @KhanAmaan55 

I personnaly use [this package](https://github.com/RomainLanz/adonis-bull-queue) to deal with redis based queues.
It's a wrapped package for AdonisJs v5 based on [BullMQ](https://github.com/taskforcesh/bullmq).

@McSneaky develop also a [queue package](https://github.com/cavai-research/Adonis-Queue) for Adonisjs but dunno if it's production-ready for the moment.

**Reply by McSneaky** [reply]
Yeah, it's in use in production at least in 3 companies so kinda production-ready

I'd go with BullMQ package when you go with Redis
The one I'm working on when you wanna go with DB (no need to mess with Redis) or for some reason write your own Redis driver :D (I haven't had time to make Redis and memory drivers yet)
Also most likely BullMQ one offers more tooling, since it's rather popular with other frameworks too

On one good day we'll turn that Adonis queue package to best queue library in whole Node ecosystem.. On that one good day when infinit time is invented 😄 (or little bit earlier)

**Reply by KhanAmaan55** [reply]
it works!! Thanks 😀
